app/controllers/api/logins_controller.rb

```diff
@@ -3,11 +3,19 @@ class Api::LoginsController < Api::ApplicationController
 
   def create
     debate = Debate.find_by code: params[:code]
-    if debate.present?
-      auth_token = debate.auth_tokens.create!
-      render json: { auth_token: auth_token.value }
+    return render json: { error: 'Debate not found' }, status: :not_found unless debate
+    form = MobileUserForm.new(::MobileUser.new)
+    if form.validate(user_params.merge(auth_token: debate.auth_tokens.create!))
+      form.save(validate: false)
+      render json: { auth_token: form.auth_token.value }
     else
-      render json: { error: 'Debate not found' }, status: :not_found
+      render json: { error: form.errors.full_messages.join('. ') }, status: :bad_request
```

Comment:
Może wyekstraktować `errors.full_messages.join('. ')` do metody formularza, wydaje się to bardziej odpowiednie miejsce do trzymania tej logiki (może da się jakoś bardziej ogólnie użyć Reform::Form)?